pkg/controller/atlasproject/backupcompliancepolicy.go

```diff
@@ -0,0 +1,151 @@
+	}()
+
+	// reference set
+	// check reference points to existing compliance policy CR
+	err := r.Client.Get(ctx.Context, *project.Spec.BackupCompliancePolicyRef.GetObject(project.Namespace), compliancePolicy)
+	if err != nil {
+		ctx.Log.Errorf("failed to get backup compliance policy: %v", err)
+		return workflow.Terminate(workflow.ProjectBackupCompliancePolicyUnavailable, err.Error())
+	}
+
+	if compliancePolicy.Annotations == nil {
+		compliancePolicy.Annotations = map[string]string{}
+	}
+	if projectIds, ok := compliancePolicy.Annotations[ProjectAnnotation]; !ok {
+		compliancePolicy.Annotations[ProjectAnnotation] = project.ID()
+	} else {
+		if !slices.Contains(strings.Split(projectIds, ","), project.ID()) {
+			compliancePolicy.Annotations[ProjectAnnotation] = projectIds + "," + project.ID()
+		}
+	}
+
+	err = r.Client.Update(ctx.Context, compliancePolicy)
+	if err != nil {
+		ctx.Log.Errorf("failed to update backup compliance policy: %v", err)
+		return workflow.Terminate(workflow.ProjectBackupCompliancePolicyUnavailable, err.Error())
+	}
+
+	// check if compliance policy exists in atlas (and matches)
+	// if match, return workflow.OK()
+	atlasCompliancePolicy, _, err := ctx.SdkClient.CloudBackupsApi.GetDataProtectionSettings(ctx.Context, project.ID()).Execute()
+
+	if err != nil {
+		ctx.Log.Errorf("failed to get backup compliance policy from atlas: %v", err)
+		return workflow.Terminate(workflow.ProjectBackupCompliancePolicyUnavailable, err.Error())
+	}
+
+	// otherwise, create/update compliance policy...
+	// create compliance policy in atlas
+	result := syncBackupCompliancePolicy(ctx, project.ID(), *compliancePolicy, *atlasCompliancePolicy)
+	ctx.SetConditionFromResult(status.BackupComplianceReadyType, result)
+	return result
+}
+
+func IsBackupComplianceEmpty(backupCompliancePolicyRef *common.ResourceRefNamespaced) bool {
+	return (backupCompliancePolicyRef == nil) || (backupCompliancePolicyRef.Name == "")
+}
+
+// syncBackupCompliancePolicy compares the compliance policy specified in Kubernetes to the one currently present in Atlas, updating Atlas should they differ.
+func syncBackupCompliancePolicy(ctx *workflow.Context, groupID string, kubeCompliancePolicy mdbv1.AtlasBackupCompliancePolicy, atlasCompliancePolicy admin.DataProtectionSettings20231001) workflow.Result {
+	// convert the CR type to atlas type, so we can compare
+	localCompliancePolicy := kubeCompliancePolicy.ToAtlas(groupID)
+	// sort the slices, so we can compare
+	if localCompliancePolicy.ScheduledPolicyItems != nil {
+		slices.SortFunc(*localCompliancePolicy.ScheduledPolicyItems, compareSPI)
+	}
+	if atlasCompliancePolicy.ScheduledPolicyItems != nil {
+		slices.SortFunc(*atlasCompliancePolicy.ScheduledPolicyItems, compareSPI)
+	}
+	// deep equal, now that the slices are sorted
+	if !equality.Semantic.DeepEqual(localCompliancePolicy, atlasCompliancePolicy) {
```

Comment:
This is comparing Atlas types, not CRD types, hence using `equality.Semantic.DeepEqual` does not make much sense here.

Didn't we want to do the reverse? I.e. always convert to our types vs. comparing Atlas types?